src/sph-partition.cc

```diff
@@ -99,16 +128,36 @@ void Worker::map(char const * const begin,
     char const * cur = begin;
     while (cur < end) {
         cur = _editor.readRecord(cur, end);
-        sc.first = _editor.get<double>(_pos.first);
-        sc.second = _editor.get<double>(_pos.second);
-        // Locate partitioning position and output a record for each location.
-        _locations.clear();
-        _chunker.locate(sc, -1, _locations);
-        assert(!_locations.empty());
-        for (LocIter i = _locations.begin(), e = _locations.end(); i != e; ++i) {
-            _editor.set(_chunkIdField, i->chunkId);
-            _editor.set(_subChunkIdField, i->subChunkId);
-            silo.add(*i, _editor);
+        if (_pos.first != -1) {
+            // RA/DEC partitioning for the director or child tables. Allowing overlaps and
+            // the "secondary" index generation (if requested).
+            sc.first = _editor.get<double>(_pos.first);
+            sc.second = _editor.get<double>(_pos.second);
+            // Locate partitioning position and output a record for each location.
+            _locations.clear();
+            _chunker.locate(sc, -1, _locations);
+            assert(!_locations.empty());
+            for (LocIter i = _locations.begin(), e = _locations.end(); i != e; ++i) {
+                _editor.set(_chunkIdField, i->chunkId);
+                _editor.set(_subChunkIdField, i->subChunkId);
+                if (!_disableChunks) silo.add(*i, _editor);
+                // Populate the "secondary" index only for the non-overlap rows.
+                if (_idField != -1 && !i->overlap) {
+                    ObjectIndex::instance().write(_editor.get(_idField, true), *i);
+                }
+            }
+        } else if (_idField != -1) {
+            // The objectId partitioning mode of a child table based on an existing
+            // "secondary" index for the FK to the corresponding "director" table.
+            std::pair<int32_t,int32_t> const chunkSubChunk = ObjectIndex::instance().read(_editor.get(_idField, true));
```

Comment:
Too many letters and some missing spaces, `auto` would work fine.

Reply:
Fixed